lte/gateway/c/oai/tasks/s1ap/s1ap_mme_nas_procedures.c

```diff
@@ -731,167 +851,204 @@ void s1ap_handle_conn_est_cnf(
    */
   //     s1ap_timer_insert(ue_ref->mme_ue_s1ap_id,
   //     ue_ref->outcome_response_timer_id);
-  message.procedureCode = S1ap_ProcedureCode_id_InitialContextSetup;
-  message.direction     = S1AP_PDU_PR_initiatingMessage;
-  initialContextSetupRequest_p =
-      &message.msg.s1ap_InitialContextSetupRequestIEs;
-  initialContextSetupRequest_p->mme_ue_s1ap_id =
-      (unsigned long) ue_ref->mme_ue_s1ap_id;
-  initialContextSetupRequest_p->eNB_UE_S1AP_ID =
-      (unsigned long) ue_ref->enb_ue_s1ap_id;
-
-  /*
-   * Only add capability information if it's not empty.
-   */
-  if (conn_est_cnf_pP->ue_radio_capability) {
-    OAILOG_DEBUG_UE(
-        LOG_S1AP, imsi64, "UE radio capability found, adding to message\n");
-    initialContextSetupRequest_p->presenceMask |=
-        S1AP_INITIALCONTEXTSETUPREQUESTIES_UERADIOCAPABILITY_PRESENT;
-    OCTET_STRING_fromBuf(
-        &initialContextSetupRequest_p->ueRadioCapability,
-        (const char*) conn_est_cnf_pP->ue_radio_capability->data,
-        conn_est_cnf_pP->ue_radio_capability->slen);
-  }
-
-  /*
-   * uEaggregateMaximumBitrateDL and uEaggregateMaximumBitrateUL expressed in
-   * term of bits/sec
-   */
+  memset(&pdu, 0, sizeof(pdu));
+  pdu.present = S1ap_S1AP_PDU_PR_initiatingMessage;
+  pdu.choice.initiatingMessage.procedureCode =
+      S1ap_ProcedureCode_id_InitialContextSetup;
+  pdu.choice.initiatingMessage.value.present =
+      S1ap_InitiatingMessage__value_PR_InitialContextSetupRequest;
+  pdu.choice.initiatingMessage.criticality = S1ap_Criticality_ignore;
+  out = &pdu.choice.initiatingMessage.value.choice.InitialContextSetupRequest;
+
+  /* mandatory */
+  ie = (S1ap_InitialContextSetupRequestIEs_t*) calloc(
```

Comment:
Where will the memory freed?

Reply:
Added "ASN_STRUCT_FREE_CONTENTS_ONLY" later in the same function